src/NUnitFramework/framework/Interfaces/ITest.cs

```diff
@@ -77,21 +77,11 @@ public interface ITest : IXmlNodeBuilder
         /// </summary>
         IMethodInfo? Method { get; }
 
-        /// <summary>
-        /// Gets the RunState of the test, indicating whether it can be run.
-        /// </summary>
-        RunState RunState { get; }
```

Comment:
If I remember right, moving interface methods down to a base interface causes MissingMethodException for already-compiled libraries that referenced the previous version of NUnit when they run against the new version of NUnit. E.g. https://stackoverflow.com/questions/55061714/missingmethodexception-after-extracting-base-interface.

The primary example is a NuGet test helper library that itself references on NUnit. The user installs the newest version of the helper library and the newest version of NUnit into their project, and then the helper library calls into `ITest.RunState` and the CLR says "I can't find that." (The CLR doesn't search base interfaces because of the diamond dependency problem. At least, this was the state of things before .NET Core added Default Interface Methods.)

Less of a hassle (because it's actionable) is that this change will also cause the following code to stop compiling:

```cs
class MyTest : ITest
{
    // ...

    // The compiler will error until ITest is changed to ICommonTestData:
    RunState ITest.RunState => _runState;
}
```

Reply:
What do you think about making `AddIgnoreUntilReason` an extension method on `IPropertyBag` so it's `test.Properties.AddIgnoreUntilReason`?